src/js/background.js

```diff
@@ -776,12 +781,45 @@ function startBackgroundListeners() {
     }
   }, {urls: ["http://*/*", "https://*/*"]}, []);
 
+  if(badger.isFirefoxMobile) {
```

Comment:
Could you refactor these to be like:
```javascript
if (!isMobile) {
  startRegularListeners(); // with an exposed named function
} else {
 startMobileListeners();
}
```

This way the mobile logic is separate from the regular logic. And so we are able to test them independently .